pathgen/pathgen.go

```diff
@@ -657,9 +659,13 @@ func (n *{{ .Struct.TypeName }}) {{ .MethodName -}} ({{ .KeyParamListStr }}) *{{
 	// because with path compression, a leaf path may be a state or config path.
 	goUnifiedLeafPathChildConstructorTemplate = mustTemplate("unifiedchildConstructor", `
 // {{ .MethodName }} corresponds to an ambiguous path; use .Config() or .State() to get a resolved path for this leaf.
-// Note: The returned struct does not implement the PathStruct interface.
 func (n *{{ .Struct.TypeName }}) {{ .MethodName -}} ({{ .KeyParamListStr }}) *{{ .ChildPkgAccessor }}{{ .TypeName }} {
 	return &{{ .ChildPkgAccessor }}{{ .TypeName }}{
+		{{ .Struct.PathBaseTypeName }}: ygnmi.New{{ .Struct.PathBaseTypeName }}(
+			[]string{ {{- .RelPathList -}} },
+			map[string]interface{}{ {{- .KeyEntriesStr -}} },
+			n,
+		),
```

Comment:
A couple concerns:
1. Does this mean all leaves are now also PathStructs and that they can be used for querying? Is this actually what we want?
2. If this is what we want, I'm wondering if they error out if two values are returned?

Reply:
1. yes and no: leaves are path structs, but path structs can't be used to query anything.
2. i don't know what you mean

Reply:
Makes sense. I misunderstood that it's the query objects that can actually be used for querying, so 2. is not relevant.